src/query.c

```diff
@@ -259,13 +259,10 @@ SDB_API char *sdb_querys (Sdb *r, char *buf, size_t len, const char *_cmd) {
 next_quote:
 		quot = strchr (quot, '"');
 		if (quot) {
-			quot--;
-			if (*quot=='\\') {
-				memmove (quot, quot + 1, strlen (quot));
-				quot++;
+			if (*(quot - 1) == '\\') {
```

Comment:
if string is "" that -1 can result on oobread.use a temporal variable to save the result of strchr and check if quot2 > quot before doing that

Reply:
line 255 `*val` is already `"`, `quot` is at least that `val` plus one, and `quot = strchr (quot, '"');` doesn't decrease `quot` (if it doesn't return NULL), so `quot-1` is guaranteed legal.